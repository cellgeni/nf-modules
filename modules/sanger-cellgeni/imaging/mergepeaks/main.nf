process IMAGING_MERGEPEAKS {
    tag "${meta.id}"
    label 'process_single'

    // conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'quay.io/cellgeni/imagetileprocessor:0.1.13'
        : 'quay.io/cellgeni/imagetileprocessor:0.1.13'}"

    input:
    tuple val(meta), val(ch_ind), path(csvs)

    output:
    tuple val(meta), path("${output_name}"), emit: merged_peaks
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_merged_peaks_ch_${ch_ind}.wkt"
    """
    merge-peaks run \
        --output_name ${output_name} \
        ${csvs} \
        ${args} \

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(merge-peaks version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_merged_peaks_ch_${ch_ind}.wkt"
    """
    touch ${output_name}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        imaging: \$(merge-peaks version)
    END_VERSIONS
    """
}
