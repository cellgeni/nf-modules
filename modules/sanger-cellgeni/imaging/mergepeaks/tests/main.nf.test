// nf-core modules test imaging/mergepeaks
nextflow_process {

    name "Test Process IMAGING_MERGEPEAKS"
    script "../main.nf"
    process "IMAGING_MERGEPEAKS"

    tag "modules"
    tag "modules_sanger"
    tag "imaging"
    tag "imaging/mergepeaks"

    test("dummy - csv ") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    0,
                    [
                        file(params.modules_testdata_base_path + 'mouse_heart_ch_0_peaks_Y0_X0.csv', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'mouse_heart_ch_0_peaks_Y0_X0_.csv', checkIfExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("dummy - csv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    0,
                    [
                        file(params.modules_testdata_base_path + 'mouse_heart_ch_0_peaks_Y0_X0.csv', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'mouse_heart_ch_0_peaks_Y0_X0_.csv', checkIfExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
