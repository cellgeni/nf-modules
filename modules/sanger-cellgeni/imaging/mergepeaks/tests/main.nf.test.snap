{
    "dummy - csv ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_0.wkt:md5,25c0054be03fcb2cd27b2165996bac90"
                    ]
                ],
                "1": [
                    "versions.yml:md5,78bffdc58887416d3a0f24aaedd12936"
                ],
                "merged_peaks": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_0.wkt:md5,25c0054be03fcb2cd27b2165996bac90"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,78bffdc58887416d3a0f24aaedd12936"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-15T01:39:31.351711898"
    },
    "dummy - csv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_0.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,78bffdc58887416d3a0f24aaedd12936"
                ],
                "merged_peaks": [
                    [
                        {
                            "id": "test"
                        },
                        "test_merged_peaks_ch_0.wkt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,78bffdc58887416d3a0f24aaedd12936"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-03-15T01:39:38.646386945"
    }
}